File: Benchmarking/src/gen_105_9.swift
  import Dispatch
  func f8(_ v0: [[Double]], _ v1: [[Double]]) -> [[Double]] {
    var v4: [[Double]] = v0
    let v3: [[[Double]]] = [v0, v4, v4, v0, v1]
    var v7: [[Double]] = v1
    let v6: [[Double]] = v3[4]
    var v8: [[Double]] = v7
    var v9: [[[Double]]] = v3
    v9[3] = v6
    var v5: [[[Double]]] = v9
    let v11: [Double] = v8[0]
    v4[1] = v11
    v9[1] = v1
    var v17: [[[Double]]] = v9
    let v20: [[Double]] = v5[4]
    v17[3] = v20
    var v42: [[[Double]]] = v5
    v9 = v5
    v9[3] = v4
    v17 = v42
    let v39: [[Double]] = v17[3]
    var v51: [[Double]] = v1
    v5[1] = v8
    v17[0] = v51
    return v39
  }
  @inline(never)
  func f0(_ v0: [[Double]], _ v1: Double) -> Double {
    let v2: [[Double]] = f8(v0, v0)
    var v5: [[Double]] = v2
    let v9: [[Double]] = f8(v0, v5)
    let v6: [Double] = v5[0]
    v5[1] = v6
    let v18: [Double] = v0[1]
    let v11: [Double] = v5[1]
    let v14: [[Double]] = [v6, v6, v11, v6]
    let v10: [[Double]] = f8(v0, v9)
    v5[1] = v6
    let v15: [[Double]] = f8(v5, v10)
    let v35: [[Double]] = f8(v9, v10)
    let v21: [[Double]] = f8(v9, v35)
    var v22: [[Double]] = v14
    let v42: [[Double]] = f8(v15, v21)
    let v33: [[Double]] = f8(v2, v42)
    let v58: [Double] = v33[1]
    let v48: Double = v58[0]
    let v47: [Double] = v22[2]
    v22[1] = v18
    let v53: [Double] = v33[0]
    v5[1] = v47
    v5[1] = v53
    return v48
  }
  func benchmark() {
    let v0: [[Double]] = [[0.0], [1.0]]
    let v1: Double = 2.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 2.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen_105_9.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  def f8(v0: Vector[Vector[Double]], v1: Vector[Vector[Double]]): Vector[Vector[Double]] = {
    var v4: Vector[Vector[Double]] = v0
    val v3: Vector[Vector[Vector[Double]]] = Vector(v0, v4, v4, v0, v1)
    var v7: Vector[Vector[Double]] = v1
    val v6: Vector[Vector[Double]] = v3(4)
    var v8: Vector[Vector[Double]] = v7
    var v9: Vector[Vector[Vector[Double]]] = v3
    v9 = v9.updated(3, v6)
    var v5: Vector[Vector[Vector[Double]]] = v9
    val v11: Vector[Double] = v8(0)
    v4 = v4.updated(1, v11)
    v9 = v9.updated(1, v1)
    var v17: Vector[Vector[Vector[Double]]] = v9
    val v20: Vector[Vector[Double]] = v5(4)
    v17 = v17.updated(3, v20)
    var v42: Vector[Vector[Vector[Double]]] = v5
    v9 = v5
    v9 = v9.updated(3, v4)
    v17 = v42
    val v39: Vector[Vector[Double]] = v17(3)
    var v51: Vector[Vector[Double]] = v1
    v5 = v5.updated(1, v8)
    v17 = v17.updated(0, v51)
    v39
  }
  @noinline
  def f0(v0: Vector[Vector[Double]], v1: Double): Double = {
    val v2: Vector[Vector[Double]] = f8(v0, v0)
    var v5: Vector[Vector[Double]] = v2
    val v9: Vector[Vector[Double]] = f8(v0, v5)
    val v6: Vector[Double] = v5(0)
    v5 = v5.updated(1, v6)
    val v18: Vector[Double] = v0(1)
    val v11: Vector[Double] = v5(1)
    val v14: Vector[Vector[Double]] = Vector(v6, v6, v11, v6)
    val v10: Vector[Vector[Double]] = f8(v0, v9)
    v5 = v5.updated(1, v6)
    val v15: Vector[Vector[Double]] = f8(v5, v10)
    val v35: Vector[Vector[Double]] = f8(v9, v10)
    val v21: Vector[Vector[Double]] = f8(v9, v35)
    var v22: Vector[Vector[Double]] = v14
    val v42: Vector[Vector[Double]] = f8(v15, v21)
    val v33: Vector[Vector[Double]] = f8(v2, v42)
    val v58: Vector[Double] = v33(1)
    val v48: Double = v58(0)
    val v47: Vector[Double] = v22(2)
    v22 = v22.updated(1, v18)
    val v53: Vector[Double] = v33(0)
    v5 = v5.updated(1, v47)
    v5 = v5.updated(1, v53)
    v48
  }
  def benchmark(): Unit = {
    val v0: Vector[Vector[Double]] = Vector(Vector(0.0), Vector(1.0))
    val v1: Double = 2.0
    val start = nanoTime()
    var result: Double = 2.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}